plugins/discovery/config.go

```diff
@@ -52,8 +53,19 @@ func (c *Config) validateAndInjectDefaults(services []string) error {
 		c.Prefix = &s
 	}
 
-	if len(services) != 1 {
-		return fmt.Errorf("discovery requires exactly one service")
+	if (c.Service == "") {
```

Comment:
Since `getServiceFromList` already has a check for `service == ""` and returning `services[0]` we could probably just move the check for `len(services) != 1` (and corresponding error) into it too and then simplify this whole section to just _always_ call `c.getServiceFromList`.

Reply:
oops, fixed that - thanks